esmvaltool/diag_scripts/eyring06jgr/eyring06jgr_fig01.ncl

```diff
@@ -0,0 +1,402 @@
+  XMin = new((/dimsizes(seasons)/), "float")
+
+  if(isatt(diag_script_info, "levmax")) then
+    Max_lev = diag_script_info@levmax
+  else
+    Max_lev = 1000.
+  end if
+  if(isatt(diag_script_info, "levmin")) then
+    Min_lev = diag_script_info@levmin
+  else
+    Min_lev = 1.
+  end if
+  if(isatt(diag_script_info, "XMax")) then
+    XMax = diag_script_info@XMax
+  else
+    XMax = 20.
+  end if
+  if(isatt(diag_script_info, "XMin")) then
+    XMin = diag_script_info@XMin
+  else
+    XMin = -30.
+  end if
+
+  multimean = diag_script_info@multimean
+  if (multimean)
+    dimTot = dim_MOD+dim_OBS
+  else
+    dimTot = dim_MOD+dim_OBS-1
+  end if
+
+  year_start = toint(diag_script_info@start_year)
+  year_end = toint(diag_script_info@end_year)
+
+  ; Create output directories
+  system("mkdir -p " + config_user_info@work_dir)
+  system("mkdir -p " + config_user_info@plot_dir)
+
+end
+
+begin
+
+  aux_title_info = ""
+  wks = get_wks("dummy_for_wks", DIAG_SCRIPT, "eyring06f01_" + var0)
+  output_filename = wks@fullname
+  nPanel = dimsizes(seasons)
+  plot = new(nPanel, "graphic")
+  plot_num = integertochar(ispan(97, 122, 1))
+  rad = 4.0 * atan(1.0) / 180.0
+  A0 = read_data(info0[iref])
+  lev0 = A0&plev
+  dimLev = dimsizes(lev0)
+  delete(A0)
+
+  tmpData = new((/dim_MOD + dim_OBS, nPanel, dimsizes(lev0), 2/), "float")
+  tmpData!0 = "line"
+  tmpData!1 = "season"
+  tmpData!2 = "plev"
+  tmpData!3 = "stat"
+
+  Xdata = new((/dimTot, nPanel, dimsizes(lev0), 2/), "float")
+  Xdata!0 = "line"
+  Xdata!1 = "season"
+  Xdata!2 = "plev"
+  Xdata!3 = "stat"
+  Xdata&plev = lev0
+
+  Xref = new((/nPanel, dimsizes(lev0), 2/), "float")
+  plotMod = new(dimTot, typeof(names))
+  plotColor = new(dimTot, typeof(names))
+  plotDash = new(dimTot, typeof(names))
+  plotThicks = new(dimTot, "float")
+  plotScatters = new(dimTot, typeof(names))
+  LineScatters = new(dimTot, typeof(names))
+  ik = 0
+
+  ; Start datasets loop
+  do imod = 0, dim_MOD1 - 1
+
+    ; Start seasons loop
+    do ipan = 0, dimsizes(seasons)-1
+
+      ; Extract data
+      A0 = read_data(info0[imod])
+      if (all(isdim(A0, (/"time", "plev", "lat", "lon"/)))) then  ; MR require?
+        dummy1 = area_operations(A0(time|:, plev|:, lat|:, lon|:), \
+                                 latrange(ipan, 0), \
+                                 latrange(ipan, 1), 0., 360., "average", True)
+        coslat = cos(A0&lat * rad)
+        coslat!0 = "lat"
+        coslat&lat = A0&lat
+        delete(coslat)
+        delete(A0)
+      else
+        if (all(isdim(A0, (/"time", "plev", "lat"/)))) then  ; MR not required!
+          coslat = cos(A0&lat * rad)
+          coslat!0 = "lat"
+          coslat&lat = A0&lat
+          dummy1 = \
+            dim_avg_wgt_Wrap(A0(:, :, {latrange(ipan, 0):latrange(ipan, 1)}), \
+                             coslat({latrange(ipan, 0):latrange(ipan, 1)}), 1)
+          delete(coslat)
+          delete(A0)
+        end if
```

Comment:
I think this whole block is redundant: do we need to distinguish the (time, plev, lat, lon) case from the (time, plev, lat) one? `ta` data in CMIP5, CMIP6 and OBS is always 4D.
This looks like a remnant from CCMVal.

Reply:
Let me try to move the zonal mean to the preprocessor. This would simplify the code a lot.